src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -600,4 +607,22 @@ private static function initFormats()
             'debug_nomax' => ' %current% [%bar%] %elapsed:6s% %memory:6s%',
         );
     }
+
+    /**
+     * @param string $line
+     * @return bool
+     */
+    private function adjustBarWidthToWindowWidth($line)
+    {
+        $lineLength = Helper::strlenWithoutDecoration($this->output->getFormatter(), $line);
+        $windowWidth = exec('tput cols');
+
+        if ($lineLength > $windowWidth) {
+            $barWidthDelta = $lineLength - $windowWidth;
+            $newBarWidth = $this->barWidth - $barWidthDelta - 20;
+            $this->setBarWidth($newBarWidth);
+            return true;
```

Comment:
An empty line required before the return... but this method looks strange to me anyway... you should change something or return something but not both... why is return required?